From 2802a34ef530c9c82e52d05dd1b6864a8b43a353 Mon Sep 17 00:00:00 2001
From: Kent McLeod <Kent.Mcleod@data61.csiro.au>
Date: Mon, 31 Jul 2017 14:29:45 +1000
Subject: [PATCH] Iperf: Remove mmap write flag

Rumprun mmap returns error if a file is mmaped r/w.  Iperf only writes
out to a file if -Z (zerocopy mode) is used.  We don't do this
currently, so this patch removes the flag
---
 src/iperf_api.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/iperf_api.c b/src/iperf_api.c
index 5b56af6..5cc0db4 100755
--- a/src/iperf_api.c
+++ b/src/iperf_api.c
@@ -2710,7 +2710,7 @@ iperf_new_stream(struct iperf_test *test, int s)
         free(sp);
         return NULL;
     }
-    sp->buffer = (char *) mmap(NULL, test->settings->blksize, PROT_READ|PROT_WRITE, MAP_PRIVATE, sp->buffer_fd, 0);
+    sp->buffer = (char *) mmap(NULL, test->settings->blksize, PROT_READ, MAP_PRIVATE, sp->buffer_fd, 0);
     if (sp->buffer == MAP_FAILED) {
         i_errno = IECREATESTREAM;
         free(sp->result);
-- 
2.11.0

